### DISCUSSION TITLE
var_sources post process

### DISCUSSION THREAD

**Discussion by schmurfy** [concourse/concourse#6400] [Help & Support]
Hello,
In my setup I have a var source defined to generate a temporary gcp key, I want to use this key with a gcs resource but...
The problem is that the json key returned by vault (by gcp to be more exact) is encoded in base64 where the gcs resource expect the key as clear text, is there currently a way to solve this ?

I want concourse to request the key as needed, that's why I use var_source, I don't see a way other than patching the gcs resource to accept base64 encoded credentials.

it it was used in my own task I could easily pre process it before usage but since it is directly used by the resource I am not sure how what to do.

Here is my resource:
```json
{
   "name": "package-user",
   "source": {
      "bucket": "packages",
      "json_key": "((gcs-key:concourse-gcs-reader.private_key_data))",
      "regexp": "user-.+",
      "unpack": true
   },
   "type": "gcs",
}
```

**Comment by taylorsilva** [comment]
I think your best option is to modify the resource :(

I can think of one potential workaround using `set_pipeline` with a pre-processing task beforehand to translate the var, but that would result in the secret being stored in the pipeline config which anyone can then see if they get `fly get-pipeline` the pipeline. Probably not what you want
